samza-core/src/test/java/org/apache/samza/zk/TestZkBarrierForVersionUpgrade.java

```diff
@@ -18,205 +18,221 @@
 import java.util.concurrent.atomic.AtomicInteger;
-//import org.junit.After;
-//import org.junit.AfterClass;
-//import org.junit.Before;
-//import org.junit.BeforeClass;
-//import org.junit.Test;
+import org.apache.samza.zk.ZkBarrierForVersionUpgrade.State;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static junit.framework.Assert.*;
+
 
 // TODO: Rename this such that it is clear that it is an integration test and NOT unit test
 public class TestZkBarrierForVersionUpgrade {
+  private static final String BARRIER_VERSION = "1";
   private static EmbeddedZookeeper zkServer = null;
   private static String testZkConnectionString = null;
   private ZkUtils zkUtils;
   private ZkUtils zkUtils1;
 
-  //@BeforeClass
+  @BeforeClass
   public static void test() {
     zkServer = new EmbeddedZookeeper();
     zkServer.setup();
-    testZkConnectionString = "127.0.0.1:" + zkServer.getPort();
+    testZkConnectionString = String.format("127.0.0.1:%d", zkServer.getPort());
   }
 
-  //@Before
+  @Before
   public void testSetup() {
     ZkClient zkClient = new ZkClient(testZkConnectionString, ZkConfig.DEFAULT_SESSION_TIMEOUT_MS, ZkConfig.DEFAULT_CONNECTION_TIMEOUT_MS);
     this.zkUtils = new ZkUtils(new ZkKeyBuilder("group1"), zkClient, ZkConfig.DEFAULT_CONNECTION_TIMEOUT_MS, new NoOpMetricsRegistry());
     ZkClient zkClient1 = new ZkClient(testZkConnectionString, ZkConfig.DEFAULT_SESSION_TIMEOUT_MS, ZkConfig.DEFAULT_CONNECTION_TIMEOUT_MS);
     this.zkUtils1 = new ZkUtils(new ZkKeyBuilder("group1"), zkClient1, ZkConfig.DEFAULT_CONNECTION_TIMEOUT_MS, new NoOpMetricsRegistry());
   }
 
-  //@After
+  @After
   public void testTearDown() {
     zkUtils.close();
     zkUtils1.close();
   }
 
-  //@AfterClass
+  @AfterClass
   public static void teardown() {
     zkServer.teardown();
   }
 
-  //@Test
-  public void testZkBarrierForVersionUpgrade() {
-    String barrierId = zkUtils.getKeyBuilder().getRootPath() + "/b1";
-    String ver = "1";
-    List<String> processors = new ArrayList<>();
-    processors.add("p1");
-    processors.add("p2");
-    final CountDownLatch latch = new CountDownLatch(2);
-    final AtomicInteger stateChangedCalled = new AtomicInteger(0);
-
-    ZkBarrierForVersionUpgrade processor1Barrier = new ZkBarrierForVersionUpgrade(barrierId, zkUtils, new ZkBarrierListener() {
-      @Override
-      public void onBarrierCreated(String version) {
-      }
+  static class TestZkBarrierListener implements ZkBarrierListener {
 
-      @Override
-      public void onBarrierStateChanged(String version, ZkBarrierForVersionUpgrade.State state) {
-        if (state.equals(ZkBarrierForVersionUpgrade.State.DONE)) {
-          latch.countDown();
-          stateChangedCalled.incrementAndGet();
-        }
-      }
+    private final CountDownLatch stateChangedLatch;
+    private final AtomicInteger stateChangedCounter;
```

Comment:
Do we need the stateChangedCounter in-addition to the stateChangedLatch? It looks like stateChangedLatch.countDown is exactly equivalent to "stateChangedCounter == 1"